### PR TITLE
Add /styleguides for git usage/pull requests

### PR DESCRIPTION
**Pull request by rgarner**
Add `/styleguides` for git usage/pull requests

Add a separate index page for style guides. Add basic git usage
guidelines and a guide for pull requests and reviews.

These guides are based very heavily on guides found at
https://github.com/alphagov/styleguides – some of the language
has changed, and there have been some updates to reflect changes
in the Github usage; removal of GDS-specific sections; and simplified
language.

Kindness and mutual respect are the core tenets.

### CODE REVIEW

File: styleguides/git.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+### Structure
+
+Start commit messages with a one-line summary no longer than 50
+characters. Various Git tools (including GitHub) use this as the commit
+summary. Format it like an email subject, with a leading capital
+and no full stop. Write it in the present tense.
+
+For example:
+
+> Add .dockerignore and tell git to ignore assets
```
Comment (sl33p) [inline styleguides/git.md]: Should this be "Adding" rather than "Add" or is that no longer a thing?

Comment (sl33p) [inline styleguides/git.md]: I see a case for present tense below. Can't recall where I got the active tense suggestion from

Comment (rgarner) [inline styleguides/git.md]: Was that a thing?

File: styleguides/pull-requests.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# Pull requests
+
+This is a pull requests style guide for working on GOV.UK.
```
Comment (rgarner) [inline styleguides/pull-requests.md]: There had to be one that slipped through :)

**Comment by rgarner** [comment]
Re: gitflow vs github flow – I expect there'll be a separate PR to this once we've [decided](https://trello.com/c/Cq8ytvzN/6-deployment) how we're going to deploy.

**Pull request merged**